File: frontend/forge-asset-web/src/types/asset.ts
export type AssetStatus = 'normal' | 'repair' | 'inactive'

export interface Asset {
  id: number
  assetCode: string
  name: string
  category: string | null
  brand: string | null
  model: string | null
  serialNumber: string | null
  location: string | null
  owner: string | null
  purchaseDate: string | null
  remark: string | null
  status: AssetStatus
  createdAt: string
  updatedAt: string
  deletedAt: string | null
}

export interface AssetFormInput {
  assetCode: string
  name: string
  category: string
  brand: string
  model: string
  serialNumber: string
  location: string
  owner: string
  purchaseDate: string | null
  remark: string
  status: AssetStatus
}

export interface GenerateAssetCodeResponse {
  assetCode: string
}

File: backend/ForgeAsset.Api/Models/Asset.cs
namespace ForgeAsset.Api.Models;

public sealed class Asset
{
    public long Id { get; set; }
    public string AssetCode { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string? Category { get; set; }
    public string? Brand { get; set; }
    public string? Model { get; set; }
    public string? SerialNumber { get; set; }
    public string? Location { get; set; }
    public string? Owner { get; set; }
    public DateTime? PurchaseDate { get; set; }
    public string? Remark { get; set; }
    public string Status { get; set; } = "normal";
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public DateTime? DeletedAt { get; set; }
}
